### PR TITLE
Fix `ArgumentError` in `original_package_update_available?` when `latest_version` is `nil`

### PR DESCRIPTION
**Pull request by kbukum1**
### What are you trying to accomplish?

The goal is to fix an `ArgumentError` that occurs when the `latest_version` is `nil` in the `original_package_update_available?` method. Instead of performing a comparison when `latest_version` is `nil`, the method will return `false`. This will prevent the error and ensure the function handles the `nil` case gracefully.

### Anything you want to highlight for special attention from reviewers?

By returning `false` when `latest_version` is `nil`, we avoid the `ArgumentError` that occurs due to the comparison with `nil`. This approach ensures that the method functions correctly even when the `latest_version` is not available.

### How will you know you've accomplished your goal?

- The error should no longer appear in the logs.
- Running the test suite should pass without errors related to this issue.
- Any additional tests should confirm that the function handles `nil` values for `latest_version` appropriately.

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb [hunk 1]
```diff
@@ -309,6 +309,8 @@ def original_package_update_available?
 
           latest_version = latest_version_finder(original_package).latest_version_from_registry
 
+          return false if latest_version.nil?
```
Comment (kbukum1) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:312]: This changes will effect the flow here, https://github.com/dependabot/dependabot-core/pull/10216/files#diff-2a34123e8f6a6891156b61d6bc345e31116149867e75b1b7148a8e9dbb35c726R111 . Needs to be confirmed if that flow is going to be ok.

**Comment by jeffwidman** [comment]
Description seems reasonable, looking forward to seeing the complete version including a simple test.

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb (hunk 1 above)
Comment (kbukum1) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb:312]: After investigating the problem, we should return `original_package_update_available` as false if `latest_version.nil?` is nil.

Regarding the following Sentry error, it is attempting to find the latest version `^1.4.4`. The caret (`^`) symbol ensures compatibility with the highest stable minor version. Therefore:

- Returning `original_package_update_available` as false makes sense.
- The check for `latest_resolvable_version` proceeds to `latest_allowable_version` and unmet_peer_dependencies, which is appropriate.
- `dependency_updates_from_full_unlock` also checks `latest_allowable_version`, which is correct.

In conclusion, we are now handling this scenario correctly.

**Comment by kbukum1** [comment]
I am having some issues when running `./spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb` also I am confused about the logic used. 

For the following test we have `latest_allowable_version: 16.3.1` but also we have `requirement: ^15.2.0`. Since current version is `15.2.0` then test is expecting not having changes which I aggree since  because of using caret, `^`. 
[Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb#L812)

But when I check the following logic we are checking if `original_package_update_available` then return otherwise we are returning  `latest_allowable_version` if `relevant_unmet_peer_dependencies` is empty. I couldn't figure out why we are returning `latest_allowable_version` if it is not fitting the requirements. 

[Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb#L303)

**Comment by kbukum1** [comment]
> I am having some issues when running `./spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb` also I am confused about the logic used.
> 
> For the following test we have `latest_allowable_version: 16.3.1` but also we have `requirement: ^15.2.0`. Since current version is `15.2.0` then test is expecting not having changes which I aggree since because of using caret, `^`. [Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb#L812)
> 
> But when I check the following logic we are checking if `original_package_update_available` then return otherwise we are returning `latest_allowable_version` if `relevant_unmet_peer_dependencies` is empty. I couldn't figure out why we are returning `latest_allowable_version` if it is not fitting the requirements.
> 
> [Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb#L303)

After doing investigation I had some confusion which are not related this PR but related how we are finding latest_resolve_version. So I think this PR is ready to go. https://github.slack.com/archives/C03UT3Q6V34/p1721242710417739

**Comment by kbukum1** [comment]
> > I am having some issues when running `./spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb` also I am confused about the logic used.
> > For the following test we have `latest_allowable_version: 16.3.1` but also we have `requirement: ^15.2.0`. Since current version is `15.2.0` then test is expecting not having changes which I aggree since because of using caret, `^`. [Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker/version_resolver_spec.rb#L812)
> > But when I check the following logic we are checking if `original_package_update_available` then return otherwise we are returning `latest_allowable_version` if `relevant_unmet_peer_dependencies` is empty. I couldn't figure out why we are returning `latest_allowable_version` if it is not fitting the requirements.
> > [Link to code](https://github.com/dependabot/dependabot-core/blob/c2b3745b92f84857ce5eb16edfc2a3f67d40f29b/npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb#L303)
> 
> After doing investigation I had some confusion which are not related directly to this PR, but related to the flow where the changes here are used in for finding latest_resolve_version, https://github.slack.com/archives/C03UT3Q6V34/p1721242710417739.
> 
> So I think this PR is ready to since the changes are reasonable for the flow.

An issue is created for clarifying the logic related to `latest_allowable_version`. Since this PR is unrelated to issue we can go further for this PR and get it done.

**Review by abdulapopoola — APPROVED** [review]
